experiment/migrator/migrator.py

```diff
@@ -0,0 +1,177 @@
+            for repo, jobs in prowConfig["postsubmits"].iteritems():
+                for (i, job) in enumerate(jobs):
+                    if prowJobName == job["name"]:
+                        prowConfig["postsubmits"][repo][i] = patchProwJob(job,
+                                                                          dashTab,
+                                                                          dashboardName)
+        #periodics -> [{name: prowjob}]
+        if "periodics" in prowConfig:
+            for i in range(0, len(prowConfig["periodics"])):
+                if prowJobName == prowConfig["periodics"][i]["name"]:
+                    prowConfig["periodics"][i] = patchProwJob(prowConfig["periodics"][i],
+                                                              dashTab,
+                                                              dashboardName)
+        #Dump ProwConfig to prowJobFile
+        prowJobFile.seek(0)
+        ruamel.yaml.dump(prowConfig, prowJobFile, Dumper=ruamel.yaml.RoundTripDumper)
+        prowJobFile.truncate()
+        prowJobFile.close()
+        #delete tab
+        dashboard["dashboard_tab"].remove(dashTab)
+
+
+def assertTabKeys(tab):
+    """Assert that the dashboard tab contains only the following keys:
+    name, description, test_group_name, alert_options->alert_mail_to_addresses
+    """
+    failingKeys = [st for st in tab.keys() if st != "name"
+                   and st != "description"
+                   and st != "test_group_name"
+                   and st != "alert_options"]
+    if failingKeys:
+        return False
+
+    if "alert_options" in tab:
+        alertKeys = tab["alert_options"].keys()
+        if len(alertKeys) != 1 or alertKeys[0] != "alert_mail_to_addresses":
+            return False
+
+    return True
+
+
+def findProwJob(name, path):
+    """Returns the path of a file that contains the named prow job
+    or "" if there isn't one. Dives into subdirectories.
+    """
+    pattern = re.compile("name: '?\"?" + name + "'?\"?$", re.MULTILINE)
+    for (dirpath, _, filenames) in walk(path):
+        for filename in filenames:
+            if filename.endswith(".yaml"):
+                for _, line in enumerate(open(dirpath + "/" + filename)):
+                    for _ in re.finditer(pattern, line):
+                        #print "Found %s in %s" % (name, filename)
+                        return dirpath + "/" + filename
+    return ""
+
+
+def patchProwJob(prowYaml, dashTab, dashboardName):
+    """Given a prow yaml and a compatible dashTab,
+    returns the new prow yaml with a new or amended annotation
+    """
+    if "annotations" in prowYaml:
+        #There exists an annotation; amend it
+        annotation = prowYaml["annotations"]
+        if "testgrid-dashboards" in prowYaml["annotations"]:
+            #Existing annotation is a testgrid annotation
+            #The dashboard name must come first if it's a sig-release-master-* dashboard
```

Comment:
out of curiosity, why is this?

Reply:
The email annotation applies the email to "...the tab created in the first dashboard specified in testgrid-dashboards" ([config.md](https://github.com/kubernetes/test-infra/blob/master/testgrid/config.md)), and not all dashboards.

Not having an email directly associated with a sig-release-master-* tab is a blocking failure in the config-checking tests.